### PR TITLE
Add the get /me route

### PR DESCRIPTION
**Pull request by cooper-ke**
should follow proper protocol!

### CODE REVIEW

File: backend/routes/user.py [hunk 1]
```diff
@@ -1,6 +1,26 @@
...
 from supabase import Client
 
+from backend.auth.authenticate_user import authenticate_user
+
 # Put routes for users (profile, following) here
 def UserRoutes(app: Flask, supabase: Client):
-    pass
\ No newline at end of file
+    # Get user profile data
+    @app.route("/user/<user_id>")
```
Comment (michael-brennan2005) [inline backend/routes/user.py:9]: wrong route do /me,
also specify this is a GET endpoint https://flask.palletsprojects.com/en/stable/quickstart/#http-methods

File: backend/routes/user.py [hunk 2]
```diff
@@ -1,6 +1,26 @@
...
+    @authenticate_user(supabase)
+    def get_me(user_id):
```
Comment (michael-brennan2005) [inline backend/routes/user.py:11]: do get_user

File: backend/routes/user.py [hunk 3]
```diff
@@ -1,6 +1,26 @@
...
+        try:
+            response = (
+                supabase.table("Users")
```
Comment (michael-brennan2005) [inline backend/routes/user.py:14]: - Table is now users
- With how auth works, you shouldnt be looking for where user_id == passed in user id, but where supabase_id == passed in supabase id

File: backend/routes/user.py [hunk 4]
```diff
@@ -1,6 +1,26 @@
...
+                .select("*")
+                .eq("user_id", user_id)
+                .execute()
+            )
+
+            if not response.data:
+                return jsonify({"error": "User not found"}), 404
+
+            return jsonify({"user": response.data[0]}), 200
```
Comment (michael-brennan2005) [inline backend/routes/user.py:23]: I think for this just return response.data[0], dont do a user key thing

File: backend/routes/user.py [hunk 5]
```diff
@@ -1,6 +1,26 @@
...
+
+        except Exception as e:
+            return jsonify({"error": str(e)}), 400
```
Comment (michael-brennan2005) [inline backend/routes/user.py:26]: Should be a 500 error

Like the error handling with this though good call

**Review by michael-brennan2005 — CHANGES_REQUESTED** [review]

File: backend/routes/user.py (hunk 2 above)
Comment (cooper-ke) [reply backend/routes/user.py]: done

File: backend/routes/user.py (hunk 3 above)
Comment (cooper-ke) [reply backend/routes/user.py]: fixed capialization, updated to .eq("supabase_id", user_id) -- but should I change the name of user_id to supabase_id?

File: backend/routes/user.py (hunk 4 above)
Comment (cooper-ke) [reply backend/routes/user.py]: done

File: backend/routes/user.py (hunk 5 above)
Comment (cooper-ke) [reply backend/routes/user.py]: done

File: backend/routes/user.py (hunk 2 above)
Comment (cooper-ke) [reply backend/routes/user.py]: well, changed get_me => get_user, is @authenticate_user correct?

**Pull request merged**